wallet/tickets.go

```diff
@@ -156,9 +166,44 @@ func (w *Wallet) VoteBitsForTicket(ticketHash *chainhash.Hash) (uint16, error) {
 
 // SetVoteBitsForTicket sets the per-ticket vote bits.  These vote bits override
 // the wallet's default vote bits.
-func (w *Wallet) SetVoteBitsForTicket(ticketHash *chainhash.Hash, voteBits uint16) error {
+func (w *Wallet) SetVoteBitsForTicket(update TicketVoteBitsUpdate) error {
+	// Sanity check for the extended voteBits length.
+	if len(update.VoteBits.ExtendedBits) > stake.MaxSingleBytePushLength-2 {
+		return fmt.Errorf("bad extended votebits length (got %v, max %v)",
+			len(update.VoteBits.ExtendedBits), stake.MaxSingleBytePushLength-2)
+	}
+
+	return walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		stakemgrNs := tx.ReadWriteBucket(wstakemgrNamespaceKey)
+		return w.StakeMgr.UpdateSStxVoteBits(stakemgrNs, &update.TicketHash,
+			update.VoteBits)
+	})
+}
+
+// SetVoteBitsForTickets sets vote bits for many given tickets.  These vote bits
+// override the wallet's default vote bits.
+func (w *Wallet) SetVoteBitsForTickets(updates []TicketVoteBitsUpdate) error {
```

Comment:
The only caller of this function already has two other slices, one of ticket hashes and another of stake.VoteBits.  I think it would make the most sense to remove this "update" type and simply pass both slices in.  Add a check that they are the same length.